### PR TITLE
Nuke the old stats folder, basic Metric SDK.

### PR DESCRIPTION
**Pull request by cijothomas**
Part 1:
Nuking stats folder is done to unblock @tarekgh 
The deleted Metric exporters will be brought back with new metrics API.

Added basic Metric SDK. This will be followed by subsequent PRs.

### CODE REVIEW

File: src/OpenTelemetry/Metrics/CounterHandleSDK.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenTelemetry.Context;
+using OpenTelemetry.Metrics.Export;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Metrics
+{
+    public class CounterHandleSDK<T> : CounterHandle<T>
```
Comment (tarekgh) [inline src/OpenTelemetry/Metrics/CounterHandleSDK.cs]: >class [](start = 11, length = 5)

would it be better to convert this to a struct?

File: src/OpenTelemetry/Metrics/CounterHandleSDK.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        public CounterHandleSDK()
+        {
+            if (typeof(T) != typeof(long) && typeof(T) != typeof(double))
+            {
+                throw new Exception("Invalid Type");
+            }
+        }
+
+        public CounterHandleSDK(string metricName, LabelSet labelset, MetricProcessor<T> metricProcessor) : this()
```
Comment (tarekgh) [inline src/OpenTelemetry/Metrics/CounterHandleSDK.cs]: >terHand [](start = 19, length = 7)

can you do the type check as you did in the other constructor?

File: src/OpenTelemetry/Metrics/CounterSDK.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        public CounterSDK()
+        {
+            if (typeof(T) != typeof(long) && typeof(T) != typeof(double))
+            {
+                throw new Exception("Invalid Type");
+            }
+        }
+
+        public CounterSDK(string name, MetricProcessor<T> metricProcessor) : this()
```
Comment (tarekgh) [inline src/OpenTelemetry/Metrics/CounterSDK.cs]: >nterSDK [](start = 18, length = 7)

type check here as you did in the other constructor.

File: src/OpenTelemetry/Metrics/Implementation/Metric.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenTelemetry.Metrics.Implementation
+{
+    public class Metric<T>
```
Comment (tarekgh) [inline src/OpenTelemetry/Metrics/Implementation/Metric.cs]: >T [](start = 24, length = 1)

just wondering, there is no any type constraints for the Metric?

File: src/OpenTelemetry/Metrics/CounterHandleSDK.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry/Metrics/CounterHandleSDK.cs]: This invokes other constructor this()

File: src/OpenTelemetry.Api/Metrics/LabelSet.cs [hunk 5]
```diff
@@ -13,12 +13,27 @@
...
     /// <summary>
     /// Normalized name value pairs of metric labels.
     /// </summary>
     public class LabelSet
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LabelSet"/> class.
+        /// </summary>
+        /// <param name="labels">labels from which labelset should be constructed.</param>
+        public LabelSet(IEnumerable<KeyValuePair<string, string>> labels)
```
Comment (tarekgh) [inline src/OpenTelemetry.Api/Metrics/LabelSet.cs]: I thought this implementation would be in the SDK and not in the APIs. @SergeyKanzhelev may advise about that.

File: src/OpenTelemetry/Metrics/CounterSDK.cs (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry/Metrics/CounterSDK.cs]: This invokes other constructor this()

File: src/OpenTelemetry.Api/Metrics/LabelSet.cs (hunk 5 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Metrics/LabelSet.cs]: You are right. It should be in SDK. I will move it out.

File: src/OpenTelemetry/Metrics/CounterSDK.cs (hunk 3 above)
Comment (tarekgh) [reply src/OpenTelemetry/Metrics/CounterSDK.cs]: You are right. I didn't notice that.

File: src/OpenTelemetry/Metrics/Implementation/Metric.cs (hunk 4 above)
Comment (cijothomas) [reply src/OpenTelemetry/Metrics/Implementation/Metric.cs]: There will be.  This part is still work in progress and i will submit follow up prs.

**Comment by tarekgh** [comment]
Thanks @cijothomas for getting this ready that quick.

**Comment by tarekgh** [comment]
> are you OK with this?

Yes, please go ahead and merge it. Thanks.

**Pull request merged**